File: kingfisher-web/src/api/dict.ts
import request from './request';

export const dictTypeApi = {
  list: (params: Record<string, unknown>) => request.get('/dict-types', { params }),
  getById: (id: number) => request.get(`/dict-types/${id}`),
  create: (data: { code: string; name: string; is_public: boolean; status: number; remark: string; version?: string }) =>
    request.post('/dict-types', data),
  update: (id: number, data: { code: string; name: string; is_public: boolean; status: number; remark: string; version?: string }) =>
    request.put(`/dict-types/${id}`, data),
  delete: (id: number) => request.delete(`/dict-types/${id}`),
  batchDelete: (ids: number[]) => request.post('/dict-types/batch-delete', { ids }),
  batchUpdateStatus: (ids: number[], status: number) => request.post('/dict-types/batch-status', { ids, status }),
};

export const dictEntryApi = {
  listByTypeId: (typeId: number, params: Record<string, unknown>) => request.get(`/dict-types/${typeId}/entries`, { params }),
  create: (typeId: number, data: { label: string; value: string; sort: number; status: number; remark: string; version?: string }) =>
    request.post(`/dict-types/${typeId}/entries`, data),
  update: (
    typeId: number,
    entryId: number,
    data: { label: string; value: string; sort: number; status: number; remark: string; version?: string },
  ) => request.put(`/dict-types/${typeId}/entries/${entryId}`, data),
  delete: (typeId: number, entryId: number) => request.delete(`/dict-types/${typeId}/entries/${entryId}`),
  batchDelete: (typeId: number, ids: number[]) => request.post(`/dict-types/${typeId}/entries/batch-delete`, { ids }),
  batchUpdateStatus: (typeId: number, ids: number[], status: number) =>
    request.post(`/dict-types/${typeId}/entries/batch-status`, { ids, status }),
  // 公共 API — 无需认证
  getPublicEntries: (code: string) => request.get(`/public/dicts/${code}/entries`),
};

File: extends/dict/domain/dict.go
package domain

import "time"

// DictType 字典类型
type DictType struct {
	ID        uint      `json:"id"`
	Code      string    `json:"code"`      // 唯一编码，如 "gender"
	Name      string    `json:"name"`      // 显示名称，如 "性别"
	IsPublic  bool      `json:"is_public"` // 是否公开（公开项可通过公共 API 读取）
	Status    int       `json:"status"`    // 1=启用, 0=禁用
	Remark    string    `json:"remark"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

// DictEntry 字典条目
type DictEntry struct {
	ID        uint      `json:"id"`
	TypeID    uint      `json:"type_id"` // 关联 dict_types.id
	Label     string    `json:"label"`   // 显示标签，如 "男"
	Value     string    `json:"value"`   // 实际值，如 "male"
	Sort      int       `json:"sort"`    // 排序
	Status    int       `json:"status"`  // 1=启用, 0=禁用
	Remark    string    `json:"remark"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}
